packages/react/src/transactions/adapters/useCreatePostController.ts

```diff
@@ -0,0 +1,69 @@
+import { CreatePost, CreatePostRequest } from '@lens-protocol/domain/use-cases/publications';
+import {
+  DelegableSigning,
+  SubsidizeOffChain,
+  SubsidizeOnChain,
+} from '@lens-protocol/domain/use-cases/transactions';
+
+import { useSharedDependencies } from '../../shared';
+import { CreatePostPresenter } from './CreatePostPresenter';
+import { CreateMomokaPostGateway } from './publications/CreateMomokaPostGateway';
+import { CreateOnChainPostGateway } from './publications/CreateOnChainPostGateway';
+
+export function useCreatePostController() {
+  const {
+    activeWallet,
+    apolloClient,
+    momokaRelayer,
+    onChainRelayer,
+    publicationCacheManager,
+    transactionFactory,
+    transactionGateway,
+    transactionQueue,
+  } = useSharedDependencies();
+
+  return async (request: CreatePostRequest) => {
+    const presenter = new CreatePostPresenter(publicationCacheManager);
+
+    const onChainGateway = new CreateOnChainPostGateway(apolloClient, transactionFactory);
+
+    const onChainPost = new SubsidizeOnChain(
+      activeWallet,
+      transactionGateway,
+      onChainGateway,
+      onChainRelayer,
+      transactionQueue,
+      presenter,
+    );
+
+    const delegableOnChainPost = new DelegableSigning(
```

Comment:
is this lens manager flow?

Reply:
correct... same as with dispatcher... the signing is delegated. No domain change in these regards, just implementation details.